### PR TITLE
Get the list of files from Autoconf vs. Git

### LINKED ISSUE #1431: CI builds fail on some platforms
Our CI builds fail on CentOS6 and Ubuntu12 because of an error occurred in check-unused-dependencies script.

The reason of the error seems that the script couldn't find git command on the platforms. I guess it's a PATH issue.

```
FAIL: tools/check-unused-dependencies
=====================================

Traceback (most recent call last):
  File "../tools/check-unused-dependencies", line 75, in <module>
    for filename in subprocess.Popen(args, stdout=subprocess.PIPE).stdout:
  File "/usr/lib64/python2.6/subprocess.py", line 642, in __init__
    errread, errwrite)
  File "/usr/lib64/python2.6/subprocess.py", line 1238, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
```

```
     74 args = ['git', 'ls-files']
     75 for filename in subprocess.Popen(args, stdout=subprocess.PIPE).stdout:
     76   filename = filename[:-1]
```

### PR DESCRIPTION
**Pull request by jablko**
This will work with source tarballs as well as checkouts.

Fixes #1431

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/freebsd-github/1494/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/linux-github/1387/ for details.

**Comment by atsci** [comment]
clang-analyzer build *successful*! See https://ci.trafficserver.apache.org/job/clang-analyzer-github/58/ for details.

File: configure.ac [hunk 1]
```diff
@@ -1937,6 +1939,9 @@ AC_CONFIG_FILES([
   tools/tsxs
 ])
 
+AC_CONFIG_FILES([tools/check-unused-dependencies],
+                [chmod +x tools/check-unused-dependencies])
```
Comment (zwoop) [inline configure.ac]: This looks odd, you are modifying the file attributes on a file owned by Git?

Comment (jablko) [inline configure.ac]: After this change, Git owns tools/check-unused-dependencies.in. But that reminds me: I should add tools/check-unused-dependencies to .gitignore ...

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/freebsd-github/1498/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/linux-github/1391/ for details.

**Comment by atsci** [comment]
clang-analyzer build *successful*! See https://ci.trafficserver.apache.org/job/clang-analyzer-github/62/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/freebsd-github/1518/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/linux-github/1412/ for details.

**Comment by atsci** [comment]
clang-analyzer build *successful*! See https://ci.trafficserver.apache.org/job/clang-analyzer-github/83/ for details.

**Pull request merged**